MetaMask/metamask-extension
# fix: UX: Multichain: Send flow: Fix "Internal JSON-RPC Error" when editing transaction from NFT to Native Currency

## **Description**

When create E2Es for the new send flow, editing a transaction from sending an NFT to sending native currency was throwing an Internal JSON-RPC Error.  After discussion with @matthewwalsh0 , using `0x` as the gas reset is the solution.

https://consensys.slack.com/archives/C03ETQA9EPK/p1713324192690549

Also adds an end to end test for ensuring that switching assets between a token to an NFT properly updates the asset.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23797?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1.  Start the new send flow
2. Choose an NFT, continue
3. Click "edit"
4. Choose the native currency 
5. Confirm the send
6. All should work

## **Screenshots/Recordings**



### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
